src/main/java/org/tdl/vireo/controller/SubmissionController.java

```diff
@@ -101,142 +106,178 @@
+        simpleSubmissionRepo.findAllBySubmitterId(user.getId()).stream().forEach(submission -> {
+            submission.setFieldValues(simpleFieldValueRepo.findAllBySubmissionId(submission.getId()));
+            submission.setLastAction(actionLogRepo.findLastEvent(submission.getId()));
+            all.add(SimpleSubmission.toSubmission(submission));
+        });
+        return new ApiResponse(SUCCESS, all);
     }
-    return new ApiResponse(SUCCESS, submission);
-  }
 
-  @RequestMapping("/advisor-review/{submissionHash}")
-  public ApiResponse getOne(@PathVariable String submissionHash) {
-    Submission submission = submissionRepo.findOneByAdvisorAccessHash(submissionHash);
-    return new ApiResponse(SUCCESS, submission);
-  }
+    @RequestMapping("/get-one/{submissionId}")
+    @PreAuthorize("hasRole('STUDENT')")
+    public ApiResponse getOne(@WeaverUser User user, @PathVariable Long submissionId) {
 
-  @RequestMapping(value = "/create", method = RequestMethod.POST)
-  @PreAuthorize("hasRole('STUDENT')")
-  public ApiResponse createSubmission(@WeaverUser User user, @WeaverCredentials Credentials credentials,
-      @RequestBody Map<String, String> data) throws OrganizationDoesNotAcceptSubmissionsException {
-
-    Submission submission = submissionRepo.create(
-      user,
-      organizationRepo.read(Long.valueOf(data.get("organizationId"))),
-      submissionStatusRepo.findByName(STARTING_SUBMISSION_STATUS_NAME),
-      credentials,
-      customActionDefinitionRepo.findAll()
-    );
-
-    return new ApiResponse(SUCCESS, submission);
-  }
+        SimpleSubmission submission = (user.getRole().ordinal() <= Role.ROLE_REVIEWER.ordinal()) 
+            ? simpleSubmissionRepo.findById(submissionId)
+            : simpleSubmissionRepo.findBySubmitterIdAndId(user.getId(), submissionId);
 
-  @Transactional
-  @RequestMapping("/delete/{submissionId}")
-  @PreAuthorize("hasRole('STUDENT')")
-  public ApiResponse deleteSubmission(@WeaverUser User user, @PathVariable Long submissionId) {
-    Submission submissionToDelete = submissionRepo.read(submissionId);
+        if (submission == null) {
+            return new ApiResponse(ERROR, "Submission not found");
+        }
 
-    ApiResponse response = new ApiResponse(SUCCESS);
-    if (submissionToDelete.getSubmitter().getEmail().equals(user.getEmail())
-        || user.getRole().ordinal() <= Role.ROLE_MANAGER.ordinal()) {
-      submissionRepo.delete(submissionToDelete);
-    } else {
-      response = new ApiResponse(ERROR, "Insufficient permisions to delete this submission.");
+        submission.setFieldValues(simpleFieldValueRepo.findAllBySubmissionId(submission.getId()));
+        submission.setActionLogs(actionLogRepo.findAll(submission.getId()));
+        submission.setSubmissionWorkflowSteps(submissionWorkflowStepRepo.findBySubmissionId(submission.getId()));
```

Comment:
Submission workflow steps are ordered per submission using `OrderColumn` annotation on the model. The `findBySubmissionId` method will have to order the same.